tests/by-util/test_nl.rs

```diff
@@ -800,3 +800,102 @@ fn test_file_with_non_utf8_content() {
         String::from_utf8_lossy(invalid_utf8)
     ));
 }
+
+// Regression tests for issue #9132: repeated flags should use last value
+#[test]
+fn test_repeated_body_numbering_flag() {
+    // -ba -bt should use -bt (number all lines)
+    new_ucmd!()
+        .args(&["-ba", "-bt"])
+        .pipe_in("a\nb\nc")
+        .succeeds()
+        .stdout_is("     1\ta\n     2\tb\n     3\tc\n");
+}
+
+#[test]
+fn test_repeated_header_numbering_flag() {
+    // -ha -ht should use -ht (number all lines in header)
+    new_ucmd!()
+        .args(&["-ha", "-ht"])
+        .pipe_in("a\nb\nc")
+        .succeeds()
+        .stdout_is("     1\ta\n     2\tb\n     3\tc\n");
+}
+
+#[test]
+fn test_repeated_footer_numbering_flag() {
+    // -fa -ft should use -ft (number all lines in footer)
+    new_ucmd!()
+        .args(&["-fa", "-ft"])
+        .pipe_in("a\nb\nc")
+        .succeeds()
+        .stdout_is("     1\ta\n     2\tb\n     3\tc\n");
+}
+
+#[test]
+fn test_repeated_number_format_flag() {
+    // -n ln -n rn should use -n rn (right-aligned)
+    new_ucmd!()
+        .args(&["-n", "ln", "-n", "rn"])
+        .pipe_in("a\nb\nc")
+        .succeeds()
+        .stdout_is("     1\ta\n     2\tb\n     3\tc\n");
+}
+
+#[test]
+fn test_repeated_number_separator_flag() {
+    // -s ':' -s '|' should use -s '|'
+    new_ucmd!()
+        .args(&["-s", ":", "-s", "|"])
+        .pipe_in("a\nb\nc")
+        .succeeds()
+        .stdout_is("     1|a\n     2|b\n     3|c\n");
+}
+
+#[test]
+fn test_repeated_number_width_flag() {
+    // -w 3 -w 8 should use -w 8
+    new_ucmd!()
+        .args(&["-w", "3", "-w", "8"])
+        .pipe_in("a\nb\nc")
+        .succeeds()
+        .stdout_is("       1\ta\n       2\tb\n       3\tc\n");
+}
+
+#[test]
+fn test_repeated_line_increment_flag() {
+    // -i 1 -i 5 should use -i 5
+    new_ucmd!()
+        .args(&["-i", "1", "-i", "5"])
+        .pipe_in("a\nb\nc")
+        .succeeds()
+        .stdout_is("     1\ta\n     6\tb\n    11\tc\n");
+}
+
+#[test]
+fn test_repeated_starting_line_number_flag() {
+    // -v 1 -v 10 should use -v 10
+    new_ucmd!()
+        .args(&["-v", "1", "-v", "10"])
+        .pipe_in("a\nb\nc")
+        .succeeds()
+        .stdout_is("    10\ta\n    11\tb\n    12\tc\n");
+}
+
+#[test]
+fn test_repeated_join_blank_lines_flag() {
+    // -l 1 -l 2 should use -l 2 (succeeds, not rejected)
```

Comment:
The end of the comment doesn't make sense.
```suggestion
    // -l 1 -l 2 should use -l 2
```